screen/src/androidMain/kotlin/composex/screen/PxSize.kt

```diff
@@ -1,15 +1,40 @@
 package composex.screen
 
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.unit.IntSize
+import androidx.window.layout.WindowMetricsCalculator
+
 
-// TODO: refactor to use WindowMetricsCalculator.computeCurrentWindowMetrics instead
 @Composable
 actual fun rememberScreenSizeInPx(): IntSize {
-    val configuration = LocalConfiguration.current
-    val dpi = configuration.densityDpi
-    val width = configuration.screenWidthDp * dpi
-    val height = configuration.screenHeightDp * dpi
-    return IntSize(width, height)
+    val context = LocalContext.current
+    val density = LocalDensity.current
+
+    val activity = context.findActivity()
+
+    return if (activity != null) {
+        val metrics = WindowMetricsCalculator.getOrCreate().computeCurrentWindowMetrics(activity)
+        val bounds = metrics.bounds
+        IntSize(bounds.width(), bounds.height())
+    } else {
+        val size = rememberScreenSizeInDp()
+        with(density) {
+            IntSize(
+                width = size.width.toPx().toInt(),
+                height = size.height.toPx().toInt()
+            )
```

Comment:
since `rememberScreenSizeInDp` is also expensive, consider using the `LocalConfiguration.current` (which is not as expensive as rememberScreenSizeInDp()) when the activity is null